New tests for matrixFileHandler and hiCMatrix

Rungetf opened the pull request:
Hi everyone,

I wrote some little tests for the matrixFileHandler and started testing on hiCMatrix only accepting `.cool` and `.h5` formats.

I'll extend the tests soon, but just to check that loading or saving with the FileHandler is working I thought I open a pull request...

@joachimwolff Am I correct that the __init__() of hiCMatrix doesn't check for any other format than `.cool` and `.h5`? So currently it accepts all formats but fails with others than `.cool` and `.h5`. Anyway I marked the test with XFail, because other formats cause an error.

@joachimwolff Can you please have a short look at the tests just to see if that's more or less how you expected the tests to be. I really only tested the runability of the load (and save if possible). But I'll come to your office tomorrow anyways...

joachimwolff commented on hicmatrix/test/test_HiCMatrix.py:
```diff
@@ -1499,3 +1584,86 @@ def test_intervalListToIntervalTree(capsys):
+                            [0., 0., 0., 0., np.nan],
+                            [0., 0., 0., 0., 0.]])
+
+    nt.assert_equal(obs_exp_matrix, test_matrix)
```
Maybe use for floating values `assert_almost_equal` with a delta of 2. 
https://docs.scipy.org/doc/numpy/reference/generated/numpy.testing.assert_almost_equal.html

joachimwolff commented on hicmatrix/test/test_matrixFileHandler.py:
```diff
@@ -0,0 +1,86 @@
+
+
+def test_load_and_save_homer(capsys):
+    homer_outfile = outfile + '.homer'
```
Please split it in `test_load` and `test_save` and mark `test_save` as xfail.

joachimwolff commented on hicmatrix/test/test_matrixFileHandler.py:
```diff
@@ -0,0 +1,86 @@
+
+def test_load_and_save_h5():
+    h5_outfile = outfile + '.h5'
+
```
Same here.

joachimwolff commented on hicmatrix/test/test_matrixFileHandler.py:
```diff
@@ -0,0 +1,86 @@
+
+def test_load_and_save_hicpro():
+    hicpro_outfile = outfile + '.hicpro'
+
```
And here.

joachimwolff commented on hicmatrix/test/test_matrixFileHandler.py:
```diff
@@ -0,0 +1,86 @@
+
+
+def test_load_and_save_ginteractions():
+    pass
```
As above but mark the load function as xfail.

joachimwolff commented on hicmatrix/test/test_HiCMatrix.py:
```diff
@@ -1499,3 +1584,86 @@ def test_intervalListToIntervalTree(capsys):
+
+    chrom_list = chrom_list.tolist()
+    test_chrom_list = ['a', 'a', 'a', 'a', '', 'a', 'a', 'a', '', 'a', 'a', '', 'a',
+                       '', 'b']
```
Please ignore the line length linting. You can exclude this rule in your linter.

joachimwolff commented:
> @joachimwolff Am I correct that the init() of hiCMatrix doesn't check for any other format than .cool and .h5? So currently it accepts all formats but fails with others than .cool and .h5. Anyway I marked the test with XFail, because other formats cause an error.

Correct it only accepts cool and h5, for all others it should fail.

Rungetf commented:
@joachimwolff I split the tests into seperate tests for saving and loading and marked some of them with Xfail. I don't think there is a testfile for the ginteraction format so I skipped this test at the moment. I'll implement it as soon as possible.

@joachimwolff I also changed the equallity test for the float matrices in the hicmatrix test to an almost_equal test as you suggested.

joachimwolff commented:
@Rungetf  One test case passed but was marked as xfail. Please check what is going on and update the test case.

joachimwolff commented on hicmatrix/test/test_matrixFileHandler.py:
```diff
@@ -0,0 +1,285 @@
+    # load data
+    matrix, cut_intervals, nan_bins, distance_counts, correction_factors = fh.load()
+
+    # TODO(Frederic): figure out how to test matrix. first printed values are below.
```
Solved?

joachimwolff commented on hicmatrix/test/test_matrixFileHandler.py:
```diff
@@ -0,0 +1,285 @@
+    matrix, cut_intervals, nan_bins, distance_counts, correction_factors = fh.load()
+
+    # TODO(Frederic): figure out how to test matrix.
+    # (0, 0)	41.345793
```
Solved?

joachimwolff commented on hicmatrix/test/test_matrixFileHandler.py:
```diff
@@ -0,0 +1,285 @@
+    matrix, cut_intervals, nan_bins, distance_counts, correction_factors = fh.load()
+
+    # TODO(Frederic): figure out how to test matrix.
+
```
Same question again
